internal/hostservice/service.go

```diff
@@ -0,0 +1,41 @@
+package hostservice
+
+import (
+	"github.com/blakewilliams/remote-development-manager/internal/hostservice/clipboard"
+	"github.com/blakewilliams/remote-development-manager/internal/hostservice/open"
+)
+
+// Service is the set of functionalities made available by the server to the client.
+type Service interface {
+	clipboard.Clipboard
+	open.Opener
+}
+
+// HostService is a default implementation of Service which offers copy paste
+// and open functionality.
+type HostService struct {
+	clipboard.Clipboard
+	open.Opener
```

Comment:
```suggestion
	clipboard: clipboard.Clipboard
```

Is there a reason we want to embed the interfaces instead of name them? Since we're calling `open.Open` directly, I think we can get away without embedding `open.Opener` since we aren't assigning an `Opener` to the struct?

I also think because of the embedding here, the `Copy` and `Paste` methods might already be implemented so we could omit the explicit methods implemented below if I understand things correctly.

Reply:
I think the interface embedding is perfectly idiomatic: a `Runner` is just something that implements the things that `Opener` and `Clipboard` do, both of which are existing interfaces we want to keep for other reasons.  Repeating the methods would be redundant and create an opportunity for divergence. This isn't that different than e.g. [`io.ReadCloser` interface just being an embedding of `Reader` and `Closer`](https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/io/io.go;l=134-136).

On the other hand, I think I agree with you that using embedding in the concrete implementations does more harm than good. Naming the fields and referring to them directly makes the code clearer -- thanks for suggesting it!

I accepted your suggestions with a bit of cleanup [here](https://github.com/BlakeWilliams/remote-development-manager/compare/BlakeWilliams:47d1953...BlakeWilliams:6947c51).